include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -399,12 +400,12 @@ class __buffer_impl
 
 struct __sycl_usm_free
 {
-    sycl::queue __q;
+    std::optional<sycl::queue> __q;
 
     void
     operator()(void* __memory) const
     {
-        sycl::free(__memory, __q);
+        sycl::free(__memory, __q.value());
```

Comment:
```suggestion
        assert(__queue.has_value());
        sycl::free(__memory, __q.value());
```

Reply:
It's probably better to have `assert` call too as @danhoeflinger proposed, because if problems occur, this situation is easier to investigate that figure out who threw an exception.

Reply:
Good point about the double-checking with `value()`.  As mentioned offline, I think `assert()` + `operator*` is probably best to avoid checks in release mode, as attempting to delete with a null queue will end up with a segfault anyway pretty quickly anyway.

Reply:
I added the assert for debug builds and switched to operator* / operator-> to have a fastpath for release / no assert builds.